src/components/NavBar/NavBar.test.jsx

```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import NavBar from '.';
+
+describe('NavBar', () => {
+  beforeEach(() => {
+    render(<NavBar />);
+  });
+
+  it('renders NavBar and its elements', () => {
+    const navBar = screen.getByTestId('navBar');
+    expect(navBar.children.length).toEqual(1);
+    const navContent = screen.getByTestId('navContent');
```

Comment:
Same as above, leave the usage of `data-test-id` as the last resource. Instead, you could use `getByRole()` to assert you have those 4 elements at specific:
- A menu icon (maybe with a `data-test-id` - depending on the final render of a Dehaze Material UI component) 
- A side menu
- An input, and
- A checkbox